FeedStoreChallenge/RealmFeedStore.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  RealmFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Salah Amassi on 27/02/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+public class RealmFeedStore: FeedStore {
+	
+	let storeURL: URL?
+	
+	public init(storeURL: URL?) {
+		self.storeURL = storeURL
+	}
+	
+	public func retrieve(completion: @escaping RetrievalCompletion) {
+		do {
+			let realm = try getRealm()
+			if let cache = realm.objects(Cache.self).last {
+				completion(.found(feed: cache.feed, timestamp: cache.timestamp))
+			} else {
+				completion(.empty)
+			}
+		} catch Realm.Error.fileNotFound, Realm.Error.filePermissionDenied {
+			completion(.empty)
+		} catch {
+			completion(.failure(error))
+		}
+	}
+	
+	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		do {
+			let realm = try getRealm()
+			try realm.write {
+				realm.add(Cache(realmFeed: feed.toRealmList(), timestamp: timestamp))
```

Comment:
Since we're always *overriding* the existing cache, the implementation should delete the existing cache *before* inserting the new values. Otherwise, it'll use too much unnecessary disk space.

```suggestion
				realm.deleteAll()
				realm.add(Cache(realmFeed: feed.toRealmList(), timestamp: timestamp))
```